### PR TITLE
Update status of children Jobs in Presubmt for non success parent job

### PR DESCRIPTION
**Pull request by sebastienvas**
In Istio, we use the RunAfterSuccess for presubmit. Currently when a job is restarted by the submit queue, once the parent job succeeds, the PR gets merged as the children are still marked as successful. This PR will update children job when parent job is updated.

### CODE REVIEW

File: prow/report/report.go [hunk 1]
```diff
@@ -52,12 +84,7 @@ func Report(ghc GithubClient, configAgent configAgent, pj kube.ProwJob) error {
 	if len(refs.Pulls) != 1 {
 		return fmt.Errorf("prowjob %s has %d pulls, not 1", pj.Metadata.Name, len(refs.Pulls))
 	}
-	if err := ghc.CreateStatus(refs.Org, refs.Repo, refs.Pulls[0].SHA, github.Status{
-		State:       string(pj.Status.State),
-		Description: pj.Status.Description,
-		Context:     pj.Spec.Context,
-		TargetURL:   pj.Status.URL,
-	}); err != nil {
+	if err := reportStatus(ghc, pj, parentJobChanged + pj.Status.Description); err != nil {
```
Comment (kargakis) [inline prow/report/report.go]: This description only makes sense if you have children jobs.

Comment (kargakis) [reply prow/report/report.go]: Nvmd, I see you set it only for children.

**Comment by kargakis** [comment]
changes lgtm

verify-gofmt fails and you also need to bump plank and jenkins-operator.

**Comment by sebastienvas** [comment]
I am currently testing this on our cluster.
https://github.com/istio/mixer/pull/1211

**Comment by BenTheElder** [comment]
This lgtm, but now has bump conflicts I think. ¯\_(ツ)_/¯

**Comment by kargakis** [comment]
ugh, needed a squash

**Comment by BenTheElder** [comment]
I did a squash and merge on github.

**Comment by sebastienvas** [comment]
on the submit queue, in istio we actually use squash instead of merge.

**Comment by kargakis** [comment]
> I did a "squash and merge" on github.

Ok, cool. Note though that for signed commits this is bad because the signature is dropped.

**Comment by BenTheElder** [comment]
> Ok, cool. Note though that for signed commits this is bad because the signature is dropped.

Right. I wouldn't do it for that, but for unsigned commits it does a pretty good job.

**Pull request merged**